packages/cinterop/src/commonMain/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -46,26 +43,80 @@ expect val INVALID_PROPERTY_KEY: PropertyKey
+interface RealmAppConfigT : CapiT
+interface RealmSyncConfigT : CapiT
+interface RealmSyncClientConfigT : CapiT
+interface RealmCredentialsT : CapiT
+interface RealmUserT : CapiT
+interface RealmNetworkTransportT : CapiT
+interface RealmSyncSessionT : CapiT
+// Public type aliases binding to internal verbose type safe type definitions. This should allow us
+// to easily change implementation details later on.
+typealias RealmAppPointer = NativePointer<RealmAppT>
+typealias RealmAppConfigurationPointer = NativePointer<RealmAppConfigT>
+typealias RealmSyncConfigurationPointer = NativePointer<RealmSyncConfigT>
+typealias RealmSyncClientConfigurationPointer = NativePointer<RealmSyncClientConfigT>
+typealias RealmCredentialsPointer = NativePointer<RealmCredentialsT>
+typealias RealmUserPointer = NativePointer<RealmUserT>
+typealias RealmNetworkTransportPointer = NativePointer<RealmNetworkTransportT>
+typealias RealmSyncSessionPointer = NativePointer<RealmSyncSessionT>
+
 @Suppress("FunctionNaming", "LongParameterList")
 expect object RealmInterop {
-    fun realm_get_version_id(realm: NativePointer): Long
+    fun realm_get_version_id(realm: RealmPointer): Long
     fun realm_get_library_version(): String
-    fun realm_get_num_versions(realm: NativePointer): Long
+    fun realm_get_num_versions(realm: RealmPointer): Long
 
-    fun realm_schema_new(schema: List<Pair<ClassInfo, List<PropertyInfo>>>): NativePointer
+    fun realm_schema_new(schema: List<Pair<ClassInfo, List<PropertyInfo>>>): RealmSchemaPointer
 
-    fun realm_config_new(): NativePointer
-    fun realm_config_set_path(config: NativePointer, path: String)
-    fun realm_config_set_schema_mode(config: NativePointer, mode: SchemaMode)
-    fun realm_config_set_schema_version(config: NativePointer, version: Long)
-    fun realm_config_set_schema(config: NativePointer, schema: NativePointer)
-    fun realm_config_set_max_number_of_active_versions(config: NativePointer, maxNumberOfVersions: Long)
-    fun realm_config_set_encryption_key(config: NativePointer, encryptionKey: ByteArray)
-    fun realm_config_get_encryption_key(config: NativePointer): ByteArray?
-    fun realm_config_set_should_compact_on_launch_function(config: NativePointer, callback: CompactOnLaunchCallback)
-    fun realm_config_set_migration_function(config: NativePointer, callback: MigrationCallback)
+    fun realm_config_new(): RealmConfigurationPointer
+    fun realm_config_set_path(config: RealmConfigurationPointer, path: String)
+    fun realm_config_set_schema_mode(config: RealmConfigurationPointer, mode: SchemaMode)
+    fun realm_config_set_schema_version(config: RealmConfigurationPointer, version: Long)
+    fun realm_config_set_schema(config: RealmConfigurationPointer, schema: RealmSchemaPointer)
+    fun realm_config_set_max_number_of_active_versions(config: RealmConfigurationPointer, maxNumberOfVersions: Long)
+    fun realm_config_set_encryption_key(config: RealmConfigurationPointer, encryptionKey: ByteArray)
```

Comment:
These are also available on synchronized Realms.

Reply:
Common properties are set on the `RealmConfigurationPointer` with the `realm_config_x`. All sync related things are separated into the sync configuration parts pointed to by `RealmSyncConfigurationPointer`.